### PR TITLE
Ensure Category attribute works well when UseSystemResourceKeys feature is turned on

### LINKED ISSUE #49554: ComponentModel Category attribute with UseSystemResourceKeys is broken
Publish and run the following application with and without UseSystemResourceKeys being set:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

</Project>
```

```C#
using System;
using System.ComponentModel;

namespace BugRepro
{
    class Program
    {
        static void Main()
        {
            Console.WriteLine(GetEnumCategory(AnEnum.Somthing));
        }
        
        public static string GetEnumCategory<T>(T enumValue)
            where T : struct, IConvertible
        {
            if (!typeof(T).IsEnum)
                return null;
            var enumCategory = enumValue.ToString();
            var fieldInfo = enumValue.GetType().GetField(enumValue.ToString());
            if (fieldInfo != null)
            {
                var attrs = fieldInfo.GetCustomAttributes(typeof(CategoryAttribute), false);
                if (attrs != null && attrs.Length > 0)
                {
                    enumCategory = ((CategoryAttribute)attrs[0]).Category;
                }
            }
            return enumCategory;
        }
    }

    public enum AnEnum
    {
        [Category("SomthingCat")]
        Somthing = 52,
    }
}
```

```
❯ dotnet publish -r win-x64 -p:PublishTrimmed=true -p:TrimMode=link -p:UseSystemResourceKeys=false
❯ C:\DotNetTest\BugRepro\bin\Debug\net5.0\win-x64\publish\BugRepro.exe
SomthingCat

❯ dotnet publish -r win-x64 -p:PublishTrimmed=true -p:TrimMode=link -p:UseSystemResourceKeys=true
❯ C:\DotNetTest\BugRepro\bin\Debug\net5.0\win-x64\publish\BugRepro.exe
PropertyCategorySomthingCat
```

_______________________

The reason this happens is because of this code:

https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs#L170-L182

If `GetLocalizedString` returns `null`, `Category` just returns what was passed to the ctor. But if `GetLocalizedString` doesn't return `null`, it uses what came from `GetLocalizedString`. Which looks like:

https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs#L200

This is looking for resources in our assembly named `PropertyCategoryXXX` where `XXX` is the category name. For example:

https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.ComponentModel.Primitives/src/Resources/Strings.resx#L120-L125

One option is to turn `UseSystemResourceKeys` off for this assembly. Or we could try to change the code somehow to do something different when `UseSystemResourceKeys=true`.

### PR DESCRIPTION
**Pull request by joperezr**
Fixes https://github.com/dotnet/runtime/issues/49554

These changes will make it so that Category attribute will not append a "PropertyCategory" as a suffix of the Category when UseSystemResourceKeys feature switch is turned on (in a trimmed application).

### CODE REVIEW

File: src/libraries/System.ComponentModel.Primitives/tests/TrimmingTests/System.ComponentModel.Primitives.TrimmingTests.proj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<Project DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props))" />
+
+  <ItemGroup>
+    <TestConsoleAppSourceFiles Include="VerifyCategoryNamesDontGetTrimmed.cs">
+      <!-- Setting the Trimming feature switch to make sure that the Resources get trimmed by the linker
+      as this test will ensure Category names will use default values whe the switch is on. -->
```
Comment (eerhardt) [inline src/libraries/System.ComponentModel.Primitives/tests/TrimmingTests/System.ComponentModel.Primitives.TrimmingTests.proj:7]:
```suggestion
      as this test will ensure Category names will use default values when the switch is on. -->
```

File: src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs [hunk 2]
```diff
@@ -199,7 +199,31 @@ public string Category
...
+            "Format" => SR.PropertyCategoryFormat,
+            "Key" => SR.PropertyCategoryKey,
+            "Layout" => SR.PropertyCategoryLayout,
+            "List" => SR.PropertyCategoryList,
+            "Mouse" => SR.PropertyCategoryMouse,
+            "Position" => SR.PropertyCategoryPosition,
+            "Scale" => SR.PropertyCategoryScale,
+            "Text" => SR.PropertyCategoryText,
+            "WindowStyle" => SR.PropertyCategoryWindowStyle,
+            _ => SR.GetResourceString("PropertyCategory" + value, value)
```
Comment (joperezr) [inline src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs:225]: Seems like adding a default value on the default case is breaking some unit tests as this is actually changing the behavior for custom categories. I will instead change the default case to not have a default value

File: src/libraries/System.ComponentModel.Primitives/tests/TrimmingTests/VerifyCategoryNamesDontGetTrimmed.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    /// When UseSystemResourceStrings feature switch is on, we want to validate that getting the resource string of
+    /// a category attribute won't result in having "PropertyCategory" appended to the beginning of the resulting string.
+    /// This test ensures that both built-in categories as well as custom categories get the right Category when the
+    /// feature switch is on.
+    /// </summary>
+    public class Program
+    {
+        public static int Main()
+        {
+            if (GetEnumCategory(AnEnum.Action) == "Action" && GetEnumCategory(AnEnum.Something) == "Something")
```
Comment (eerhardt) [inline src/libraries/System.ComponentModel.Primitives/tests/TrimmingTests/VerifyCategoryNamesDontGetTrimmed.cs:19]: Can you add a test for one of the strings with a space in it? Like `WindowStyle`? That will ensure we aren't just returning the string back, but instead getting the real "resource" value.

**Review by eerhardt — APPROVED** [review]
LGTM

**Pull request merged**